From 815573851dab5b907fe3076b0272fb1c03a2b245 Mon Sep 17 00:00:00 2001
From: nx111 <NX111.AimH@gmail.com>
Date: Thu, 6 Dec 2018 20:49:09 +0800
Subject: [PATCH 2/3] Revert "LineageParts: Drop `Control ringtone volume`
 setting"

This reverts commit ed5801cfadd13cb05d7a2184e528ee9fa5fad0e6.

Change-Id: Id529f834b37204373a010a90b5709c675ba7a161
---
 res/values/strings.xml                                   | 4 ++++
 res/xml/button_settings.xml                              | 7 +++++++
 src/org/lineageos/lineageparts/input/ButtonSettings.java | 1 +
 3 files changed, 12 insertions(+)

diff --git a/res/values/strings.xml b/res/values/strings.xml
index 0eef325..f44e749 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -260,6 +260,10 @@
     <string name="power_menu_bug_report_disabled">Bug reporting is disabled in development settings</string>
     <string name="power_menu_lockdown_title">Lockdown</string>
 
+    <string name="volume_keys_control_ring_stream_title">Control ringtone volume</string>
+    <string name="volume_keys_control_ring_stream_summary_on">Volume keys control ringtone volume</string>
+    <string name="volume_keys_control_ring_stream_summary_off">Volume keys control media volume</string>
+
     <string name="camera_double_tap_power_gesture_title">Press power button twice for camera</string>
 
     <!-- Description of setting that controls gesture to open camera by double tapping the power button [CHAR LIMIT=NONE] -->
diff --git a/res/xml/button_settings.xml b/res/xml/button_settings.xml
index 2104a65..bf15b56 100644
--- a/res/xml/button_settings.xml
+++ b/res/xml/button_settings.xml
@@ -275,6 +275,13 @@
             android:summary="@string/volbtn_music_controls_summary"
             android:defaultValue="true" />
 
+        <lineageos.preference.LineageSystemSettingSwitchPreference
+            android:key="volume_keys_control_ring_stream"
+            android:title="@string/volume_keys_control_ring_stream_title"
+            android:summaryOn="@string/volume_keys_control_ring_stream_summary_on"
+            android:summaryOff="@string/volume_keys_control_ring_stream_summary_off"
+            android:defaultValue="true" />
+
         <ListPreference
             android:key="volume_key_cursor_control"
             android:dialogTitle="@string/volbtn_cursor_control_title"
diff --git a/src/org/lineageos/lineageparts/input/ButtonSettings.java b/src/org/lineageos/lineageparts/input/ButtonSettings.java
index 9b3b013..fefa935 100644
--- a/src/org/lineageos/lineageparts/input/ButtonSettings.java
+++ b/src/org/lineageos/lineageparts/input/ButtonSettings.java
@@ -79,6 +79,7 @@ public class ButtonSettings extends SettingsPreferenceFragment implements
     private static final String KEY_POWER_END_CALL = "power_end_call";
     private static final String KEY_HOME_ANSWER_CALL = "home_answer_call";
     private static final String KEY_VOLUME_MUSIC_CONTROLS = "volbtn_music_controls";
+    private static final String KEY_VOLUME_CONTROL_RING_STREAM = "volume_keys_control_ring_stream";
     private static final String KEY_TORCH_LONG_PRESS_POWER_GESTURE =
             "torch_long_press_power_gesture";
     private static final String KEY_TORCH_LONG_PRESS_POWER_TIMEOUT =
-- 
2.17.1

